checker/checker_test.go

```diff
@@ -183,6 +183,63 @@ func TestCompile(t *testing.T) {
 		}
 		`,
 		nil,
+	}, {
+		"errors with duplicate function names",
+		`
+		fs duplicateFunctionName() {}
+		fs duplicateFunctionName() {}
+		`,
+		ErrDuplicateDecls{},
```

Comment:
Can we modify how we test in order to ensure that the `err.Error()` matches what's returned by `checker.Check`?

Reply:
Sounds good. I have updated the tests to have all the information needed to validate `err.Error()`. 

Do you think the position of the error is relevant? or is it ok to remove? Seems like the position would be very brittle and require re-skinning of test if, for example, the formatter would change.